### ISSUE TITLE
zellij just crashed

### ISSUE THREAD

**Issue by xxx** [zellij-org/zellij#1284]
Hi, zellij just crashed while I was interacting with a Ruby on Rails console.

**Basic information**

`zellij --version`: zellij 0.24.0
`tput lines`: 54
`tput cols`: 207
(23 lines, 101 cols, for the pane size at time of crash - above numbers are the entire screen)
`uname -av` or `ver`(Windows): Linux vatican 5.15.0-22-generic #22-Ubuntu SMP Tue Feb 8 10:16:30 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux

List of programs you interact with as, `PROGRAM --version`: output cropped meaningful, for example:
`alacritty --version`: alacritty 0.10.1 ()
`fish --version`: fish, version 3.3.1
`starship --version`: 
```
starship 1.3.0
branch:
commit_hash:
build_time:2022-02-15 17:05:27 -08:00
build_env:rustc 1.58.0 (02072b482 2022-01-11),stable-x86_64-unknown-linux-gnu
```
`ruby --version`: ruby 3.1.0p0 (2021-12-25 revision fb4df44d16) [x86_64-linux]
Ruby on Rails version: 7.0.1

**Further information**
Reproduction steps, noticeable behavior, related issues, etc

I was interacting with a Rails console, and had just pressed enter from a command. Nothing grapically unusual should have happened. I've spent hundreds of hours in Rails consoles through zellij without issue. This is the very first time anything averse has ever occurred.
One potential thing to note, is that I have disabled the autocomplete popup that comes with `irb` on Ruby 3.1.0.

full trace:

```
Error occurred in server:
Originating Thread(s):
1. screen_thread: HandlePtyBytes

Error: thread 'screen' panicked at 'called `Option::unwrap()` on a `None` value': /home/mpd/.cargo/registry/src/github.com-1ecc6299db9ec823/zellij-server-0.24.0/src/panes/grid.rs:1149
   0: zellij_utils::errors::handle_panic
             at .cargo/registry/src/github.com-1ecc6299db9ec823/zellij-utils-0.24.0/src/errors.rs:25:21
   1: std::panicking::rust_panic_with_hook
             at /rustc/59eed8a2aac0230a8b53e89d4e99d55912ba6b35/library/std/src/panicking.rs:628:17
   2: std::panicking::begin_panic_handler::{{closure}}
             at /rustc/59eed8a2aac0230a8b53e89d4e99d55912ba6b35/library/std/src/panicking.rs:519:13
   3: std::sys_common::backtrace::__rust_end_short_backtrace
             at /rustc/59eed8a2aac0230a8b53e89d4e99d55912ba6b35/library/std/src/sys_common/backtrace.rs:141:18
   4: rust_begin_unwind
             at /rustc/59eed8a2aac0230a8b53e89d4e99d55912ba6b35/library/std/src/panicking.rs:517:5
   5: core::panicking::panic_fmt
             at /rustc/59eed8a2aac0230a8b53e89d4e99d55912ba6b35/library/core/src/panicking.rs:101:14
   6: core::panicking::panic
             at /rustc/59eed8a2aac0230a8b53e89d4e99d55912ba6b35/library/core/src/panicking.rs:50:5
   7: <zellij_server::panes::grid::Grid as vte::Perform>::csi_dispatch
   8: vte::Parser::perform_state_change
             at .cargo/registry/src/github.com-1ecc6299db9ec823/vte-0.10.1/src/lib.rs:175:17
      vte::Parser::advance
             at .cargo/registry/src/github.com-1ecc6299db9ec823/vte-0.10.1/src/lib.rs:130:9
      <zellij_server::panes::terminal_pane::TerminalPane as zellij_server::tab::Pane>::handle_pty_bytes
             at .cargo/registry/src/github.com-1ecc6299db9ec823/zellij-server-0.24.0/src/panes/terminal_pane.rs:98:13
   9: zellij_server::tab::Tab::process_pty_bytes
             at .cargo/registry/src/github.com-1ecc6299db9ec823/zellij-server-0.24.0/src/tab/mod.rs:726:13
  10: zellij_server::tab::Tab::handle_pty_bytes
             at .cargo/registry/src/github.com-1ecc6299db9ec823/zellij-server-0.24.0/src/tab/mod.rs:709:9
      zellij_server::screen::screen_thread_main
             at .cargo/registry/src/github.com-1ecc6299db9ec823/zellij-server-0.24.0/src/screen.rs:707:25
  11: zellij_server::init_session::{{closure}}
             at .cargo/registry/src/github.com-1ecc6299db9ec823/zellij-server-0.24.0/src/lib.rs:613:17
      std::sys_common::backtrace::__rust_begin_short_backtrace
             at /rustc/59eed8a2aac0230a8b53e89d4e99d55912ba6b35/library/std/src/sys_common/backtrace.rs:125:18
  12: std::thread::Builder::spawn_unchecked::{{closure}}::{{closure}}
             at /rustc/59eed8a2aac0230a8b53e89d4e99d55912ba6b35/library/std/src/thread/mod.rs:481:17
      <core::panic::unwind_safe::AssertUnwindSafe<F> as core::ops::function::FnOnce<()>>::call_once
             at /rustc/59eed8a2aac0230a8b53e89d4e99d55912ba6b35/library/core/src/panic/unwind_safe.rs:271:9
      std::panicking::try::do_call
             at /rustc/59eed8a2aac0230a8b53e89d4e99d55912ba6b35/library/std/src/panicking.rs:403:40
      std::panicking::try
             at /rustc/59eed8a2aac0230a8b53e89d4e99d55912ba6b35/library/std/src/panicking.rs:367:19
      std::panic::catch_unwind
             at /rustc/59eed8a2aac0230a8b53e89d4e99d55912ba6b35/library/std/src/panic.rs:129:14
      std::thread::Builder::spawn_unchecked::{{closure}}
             at /rustc/59eed8a2aac0230a8b53e89d4e99d55912ba6b35/library/std/src/thread/mod.rs:480:30
      core::ops::function::FnOnce::call_once{{vtable.shim}}
             at /rustc/59eed8a2aac0230a8b53e89d4e99d55912ba6b35/library/core/src/ops/function.rs:227:5
  13: <alloc::boxed::Box<F,A> as core::ops::function::FnOnce<Args>>::call_once
             at /rustc/59eed8a2aac0230a8b53e89d4e99d55912ba6b35/library/alloc/src/boxed.rs:1636:9
      <alloc::boxed::Box<F,A> as core::ops::function::FnOnce<Args>>::call_once
             at /rustc/59eed8a2aac0230a8b53e89d4e99d55912ba6b35/library/alloc/src/boxed.rs:1636:9
      std::sys::unix::thread::Thread::new::thread_start
             at /rustc/59eed8a2aac0230a8b53e89d4e99d55912ba6b35/library/std/src/sys/unix/thread.rs:106:17
  14: <unknown>
  15: <unknown>
```

**Comment by auronandace** [comment]
Can you reliably reproduce this?

I think the tput information is only really useful if you also provide debug logs.

Looking through the backtrace I can say that there have been changes to those portions of code since 0.24. The latest release is 0.26.1. If you can't reliably reproduce this then it will be difficult to debug. You can try the latest release and see if the issue happens again.

**Comment by a-kenji** [comment]
This shouldn't be fixed in the current release. But I am reasonably confident that it is fixed in main and will be in the next release.

**Comment by imsnif** [comment]
Hey, thanks for the report and apologies for the experience!

The crash happened here: https://github.com/zellij-org/zellij/blob/v0.24.0/zellij-server/src/panes/grid.rs#L1149
This is a helper function we use when eg. the cursor is moved to the middle of the line and we need to pad the columns that come before it. At this helper function, we assume the line exists in our viewport and so unwrap it.

For some reason here, the state was corrupted and the cursor was in the middle of a line that does not exist. Unfortunately, the trace doesn't show us how we got to this helper function. The proper fix would be there. But barring that, I issued a small patch that would recover from this state by padding the lined until the current line in case this happens.
